backend/tests/icpy/test_youtube_download_tool.py

```
@@ -0,0 +1,209 @@
+            assert 'description' in preset
+            assert 'max_filesize' in preset
+    
+    def test_rate_limiting(self, tool):
+        """Test rate limiting mechanism"""
+        from icpy.agent.tools.youtube_download_tool import _rate_limit_tracker
+        
+        # Clear tracker
+        _rate_limit_tracker.clear()
+        
+        # First 3 requests should succeed
+        for i in range(3):
```

Ruff (0.14.14)

[warning] Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)